hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -40,14 +40,16 @@ class BaseSocket : public hazelcast::client::socket
                const address& addr,
                client::config::socket_options& socket_options,
                boost::asio::io_context& io,
-               std::chrono::milliseconds& connect_timeout_in_millis)
-      : socket_options_(socket_options)
+               std::chrono::milliseconds& connect_timeout_in_millis,
+               connection::Connection& connection)
+      : is_closed_{ true }
```

Comment:
Should we set is_closed_ to false in "void close() override" method as it is called from other classes.